src/routes/data/crypto_operation.rs

```diff
@@ -35,18 +38,44 @@ pub async fn encrypt_data_and_insert_into_db<'a>(
     Ok(locker)
 }
 
-pub async fn decrypt_data(
+pub async fn decrypt_data<T>(
     tenant_app_state: &TenantAppState,
     crypto_operator: Box<dyn CryptoOperationsManager>,
-    mut locker: Locker,
-) -> Result<Locker, ContainerError<error::ApiError>> {
-    if let Some(encrypted_data) = locker.data.get_encrypted_inner_value() {
+    mut data: T,
+) -> Result<T, ContainerError<error::ApiError>>
+where
+    T: types::SecretDataManager,
+{
+    if let Some(encrypted_data) = data.get_encrypted_inner_value() {
         let decrypted_data = crypto_operator
             .decrypt_data(tenant_app_state, encrypted_data)
             .await?;
 
-        locker.data = Encryptable::from_decrypted_data(decrypted_data)
+        data = data.set_decrypted_data(decrypted_data);
+        Ok(data)
+    } else {
+        Err(error::ApiError::NotFoundError.into())
     }
+}
```

Comment:
Why does it through a error when `get_encrypted_inner_value` fail.

Reply:
We should not return the error, if it is going to the else case means that data is already decrypted so we can just return it.